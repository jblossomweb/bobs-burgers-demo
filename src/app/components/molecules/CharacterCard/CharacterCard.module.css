.card {
  @apply h-80 text-center;
}

.info {
  @apply my-2 px-4 h-10 overflow-hidden text-sm;
}

.button {
  @apply w-full text-base;
}
