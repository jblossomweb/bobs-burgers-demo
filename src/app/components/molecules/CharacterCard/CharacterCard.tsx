import React from 'react'
import { Link } from 'react-router-dom'
import type Character from 'app/models/Character'
import Button from 'app/components/atoms/Button'
import Card from 'app/components/atoms/Card'
import CharacterAvatar from 'app/components/ions/CharacterAvatar'
import styles from './CharacterCard.module.css'

export interface Props {
  character: Character
}

const CharacterCard: React.FC<Props> = ({ character }) => (
  <Card
    className={styles.card}
    width={300}
    title={character.name}
    foot={
      <Link to={`/characters/${character.id}`}>
        <Button className={styles.button}>Details...</Button>
      </Link>
    }
  >
    <CharacterAvatar character={character} />
    <p className={styles.info}>{character.occupation}</p>
  </Card>
)

export default CharacterCard
