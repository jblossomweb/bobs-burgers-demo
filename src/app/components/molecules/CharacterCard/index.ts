export { default, type Props } from './CharacterCard'
