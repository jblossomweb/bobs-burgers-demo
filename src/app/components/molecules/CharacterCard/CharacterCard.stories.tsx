import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import mockCharacter from 'app/models/__mocks__/character.json'

import CharacterCard, { type Props } from './CharacterCard'

export default {
  title: 'components/molecules/CharacterCard',
  component: CharacterCard,
  argTypes: {
    character: { control: 'object' },
  },
} as ComponentMeta<typeof CharacterCard>

const Template: ComponentStory<typeof CharacterCard> = ({
  character,
}: Props) => <CharacterCard character={character} />

export const Sample = Template.bind({})
Sample.args = {
  character: mockCharacter,
}
