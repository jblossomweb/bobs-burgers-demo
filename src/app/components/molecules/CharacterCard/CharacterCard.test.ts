import storySnaps from 'test/utils/storySnaps'
import * as stories from './CharacterCard.stories'

describe('app/components/molecules/CharacterCard', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
