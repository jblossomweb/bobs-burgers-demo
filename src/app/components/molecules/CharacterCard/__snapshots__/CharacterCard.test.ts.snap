// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/molecules/CharacterCard stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="width: 300px;"
  >
    <div
      class="card card"
    >
      <div
        class="body"
      >
        <h2
          class="title"
        >
          Robert "Bob" Belcher, Jr.
        </h2>
        <div
          class="content"
        >
          <div
            class="avatar avatar"
            style="background-color: rgb(97, 218, 251);"
          >
            <div
              class="wrapper"
            >
              <a
                href="/characters/47"
              >
                <img
                  alt="Robert \\"Bob\\" Belcher, Jr."
                  class="image"
                  src="https://bobsburgers-api.herokuapp.com/images/characters/47.jpg"
                />
              </a>
            </div>
          </div>
          <p
            class="info"
          >
            Chef/owner and operator of Bob's Burgers
          </p>
        </div>
      </div>
      <a
        href="/characters/47"
      >
        <button
          class="button button"
          type="button"
        >
          Details...
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
